### PR TITLE
refactoring of settings classes

### PR DESCRIPTION
**Pull request by patriknw**
* rename Config classes to Settings
* all settings internal, move the public part to KeyspaceAndTableStatements
* move all settings related to eventsByTag to EventsByTagSettings
* add EventsByTagSettings
* move pubsub-notification config to events-by-tag config section
* move first-time-bucket config to events-by-tag config section

### CODE REVIEW

**Comment by patriknw** [comment]
draft because on top of other PR

**Comment by patriknw** [comment]
See #81

File: core/src/main/scala/akka/persistence/cassandra/EventsByTagSettings.scala [hunk 1]
```diff
@@ -112,78 +106,86 @@ import com.typesafe.config.Config
...
-    config: Config,
-    writePluginConfig: CassandraJournalConfig)
-    extends NoSerializationVerificationNeeded {
+@InternalApi private[akka] class EventsByTagSettings(system: ActorSystem, config: Config) {
+  import EventsByTagSettings._
+  private val eventsByTagConfig = config.getConfig("events-by-tag")
+  private val log = Logging(system, classOf[EventsByTagSettings])
 
-  val log = Logging(system, classOf[CassandraReadJournalConfig])
+  val eventsByTagEnabled: Boolean = eventsByTagConfig.getBoolean("enabled")
```
Comment (johanandren) [inline core/src/main/scala/akka/persistence/cassandra/EventsByTagSettings.scala]: What about Scala constant style vs regular field, in the Akka `Settings` classes we tend to go with the constant style. Should we align these as well when anyways messing with them?

**Comment by patriknw** [comment]
@chbatey any comments, or can I merge?

**Pull request merged**